### ISSUE TITLE
UnusedPrivateMember false positives on private extension functions

### ISSUE THREAD

**Issue by brentwatson** [detekt/detekt#9139]
## Observed Behavior

Getting `UnusedPrivateMember` failures with Kotlin 2.3.10 & detekt 1.23.8 on private Extension functions where the receiver type is from another module.

## Steps to Reproduce

Example:

```kotlin
// module1
public data class Type1(
    public val title: String,
) : Serializable

public data class Type2(
    public val title2: String,
) : Serializable


// module2
class Demo {
    fun useIt() {
       val type2 = Type1("test").toType2()
       println(type2)
    }

    private fun Type1.toType2(): Type2 = Type2(title)
}
```

Detekt fails on the above:

```bash
> ./gradlew detektMain
...
        UnusedPrivateMember - [Private function `toType2` is unused.] at /Users/...project.../Demo.kt:12:22
```

The above is a simple example. Seeing this issue a lot on a very large project. Started happening after updating from Kotlin 2.2.20 -> Kotlin 2.3.10

## Your Environment
* Version of detekt used: 1.23.8
* Version of Gradle used: 9.2.1
* Version of Kotlin used: 2.3.10
* Operating System and version: MacOS Tahoe 26.1

**Comment by atulgpt** [comment]
Hi @brentwatson can you try with the latest version of detekt?

**Comment by brentwatson** [comment]
Closing. See: https://github.com/detekt/detekt/issues/9145#issuecomment-4085047013
